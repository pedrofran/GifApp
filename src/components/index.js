export * from './addCategory/addCategory'
export * from './GidGrid/GifGrid'
export * from './GidGrid/GifItem/GifItem'