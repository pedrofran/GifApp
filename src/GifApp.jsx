import {useState} from 'react'
import { AddCategory, GifGrid } from './components'

export const GifApp = () => {
    const [categories, setCategories] = useState('')

    //const addCategory =  () => setCategories([...categories, "Asdf"])
    const onChangeCategory =  (changeCategory) => setCategories(changeCategory)

  return (
    <>
        <h1>GifApp</h1>

        <AddCategory setCategories={setCategories} 
            onChangeCategory={onChangeCategory}
        />

        
        <GifGrid category={categories}/>
    </>

  )
}
