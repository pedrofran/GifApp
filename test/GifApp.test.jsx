import { getByText, render, screen } from '@testing-library/react';
import { GifApp } from '../src/GifApp';

describe('Pruebas en <GifApp />', () => {

  test('debe hacer match con el snapshot', () => {   
    const { container } = render(<GifApp />)
    expect(container).toMatchSnapshot();
  }) 

  test('debe cambiar el valor de la caja de texto', () => { 
    
      render( <GifApp /> );

      expect(screen.getByText("GifApp"));
   })




});