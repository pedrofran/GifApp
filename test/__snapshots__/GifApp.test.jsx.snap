// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifApp /> debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    GifApp
  </h1>
  <input
    placeholder="Buscar Gifs"
    type="text"
    value=""
  />
  <h3 />
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
